libraries/AP_ToshibaCAN/AP_ToshibaCAN.cpp

```diff
@@ -313,9 +313,18 @@ void AP_ToshibaCAN::loop()
                     if (esc_id < TOSHIBACAN_MAX_NUM_ESCS) {
                         WITH_SEMAPHORE(_telem_sem);
                         _telemetry[esc_id].rpm = be16toh(reply_data.rpm);
-                        _telemetry[esc_id].voltage_mv = be16toh(reply_data.voltage_mv);
+                        _telemetry[esc_id].current_ca = MAX((int16_t)be16toh(reply_data.current_ma), 0) * (4.0f * 0.1f);    // milli-amps to centi-amps
+                        _telemetry[esc_id].voltage_cv = be16toh(reply_data.voltage_mv) * 0.1f;  // millivolts to centi-volts
                         _telemetry[esc_id].count++;
                         _telemetry[esc_id].new_data = true;
+                        // update total current
+                        const uint32_t now_ms = AP_HAL::millis();
+                        const uint32_t diff_ms = now_ms - _telemetry[esc_id].last_update_ms;
+                        if (diff_ms <= 1000) {
```

Comment:
@OXINARF, my concern was mostly that at startup that diff_ms could be very large or possibly if ESC telemetry stopped and then we would get some almost random jump in the total current.  I just think that extrapolating the current draw if the time period is more than 1 second is going to be inaccurate.  Maybe it could be 5sec or 10sec.. but I think we need some limit.  Anyway, I think I'm going to leave it like this and there's a strong chance that it never causes problems.